src/web/app/components/copy-course-modal/copy-course-modal.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { TimezoneService } from '../../../services/timezone.service';
+import { Course, FeedbackSession } from '../../../types/api-output';
+import { COURSE_ID_MAX_LENGTH } from '../../../types/field-validator';
+
+interface Timezone {
+  id: string;
+  offset: string;
+}
+
+/**
+ * Copy course modal.
+ */
+@Component({
+  selector: 'tm-copy-course-modal',
+  templateUrl: './copy-course-modal.component.html',
+  styleUrls: ['./copy-course-modal.component.scss'],
+})
+export class CopyCourseModalComponent implements OnInit {
+
+  // const
+  COURSE_ID_MAX_LENGTH: number = COURSE_ID_MAX_LENGTH;
+
+  @Input()
+  courseToFeedbackSession: Record<string, FeedbackSession[]> = {};
+
+  @Input()
+  activeCourses: Course[] = [];
+
+  @Input()
+  allCourses: Course[] = [];
+
+  isCopyFromOtherSession: boolean = false;
+  newCourseIdIsConflicting: boolean = false;
+  timezones: Timezone[] = [];
+  newTimezone: string = '';
+  newCourseId: string = '';
+  newCourseName: string = '';
+  oldCourseId: string = '';
+  oldCourseName: string = '';
+
+  selectedFeedbackSessions: Set<FeedbackSession> = new Set<FeedbackSession>();
+
+  constructor(public activeModal: NgbActiveModal,
+              private statusMessageService: StatusMessageService,
+              private timezoneService: TimezoneService) {}
+
+  ngOnInit(): void {
+    this.timezones = Object.entries(this.timezoneService.getTzOffsets())
+      .map(([id, offset]: [string, number]) => {
+        const hourOffset: number = Math.floor(Math.abs(offset) / 60);
+        const minOffset: number = Math.abs(offset) % 60;
+        const sign: string = offset < 0 ? '-' : '+';
+        return { id, offset: offset === 0 ? 'UTC' : `UTC ${sign}${zeroPad(hourOffset)}:${zeroPad(minOffset)}` };
+      });
+    this.newTimezone = this.timezoneService.guessTimezone();
+  }
+
+  /**
+   * Fires the copy event.
+   */
+  copy(): void {
+    if (!this.newCourseId || !this.newCourseName) {
+      this.statusMessageService.showErrorToast(
+          'Please make sure you have filled in both Course ID and Name before adding the course!');
+      return;
+    }
+
+    this.newCourseIdIsConflicting = this.allCourses
+      .filter((course: Course) => course.courseId === this.newCourseId).length > 0;
+    if (this.newCourseIdIsConflicting) {
+      this.statusMessageService.showErrorToast(
+        `The course ID ${this.newCourseId} has been used by another course, possibly by some other user.`);
```

Comment:
This message makes it sound like you're comparing the name with all courses in the DB, not just all courses from the user. Is it really what's being done here?

Reply:
This part checks only for courses that belong to the instructor, I will update the message.